index.js

```diff
@@ -89,4 +97,25 @@ module.exports = function (processors, options) {
   }
 
   return stream
+})
+
+
+function withConfigLoader(cb) {
+  return function (plugins, options) {
+    if (typeof plugins === 'undefined') {
+      return cb(postcssLoadConfig)
```

Comment:
> From user perspective ctx.from => ctx.dir && ctx.ext is more convenient

We're talking about advanced usage here, that should not be a concern of complicating the API

In both examples everything depends on the filename, so `modules: true | false` can be derived in `postcss.config.js`. I don't want to create a config for a config that returns a config that is then overriding the default config. This is such a complicated API. If this what folks want, it would've been easier for them to pipe through `through2` and run postcss however they want. I may as well just provide a callback for `(gulpFile) => { plugins, options }` .  Surprisingly no one has ever asked this before, so I assume this is working on features that will not be used.